benchmarking_hallucination_metrics/benchmark_hallucination_metrics.ipynb

```diff
@@ -0,0 +1,4732 @@
+        "The output should strictly use the following template: Explanation: [provide a brief reasoning you used to derive the rating Score] and then write 'Score: <rating>' on the last line.\n",
+        "\"\"\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "Through trial and error, we also found the following score extraction regexes to be useful:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 18,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "import re\n",
+        "\n",
+        "\n",
+        "regex_list = [\n",
+        "    re.compile(\n",
+        "        r\".*(?:^|\\n)\\s*score:?\\s*\\(?(?P<answer>one|two|three|four|five|[12345])\\)?\",\n",
+        "        flags=re.DOTALL | re.IGNORECASE\n",
+        "    ),\n",
+        "    re.compile(\n",
+        "        r\".*\\(?(?P<answer>one|two|three|four|five|[12345])\\)?\",\n",
+        "        flags=re.DOTALL | re.IGNORECASE\n",
+        "    )\n",
+        "]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 19,
+      "metadata": {
+        "id": "Xvoa7jVj3J9V"
+      },
+      "outputs": [],
+      "source": [
+        "import numpy as np\n",
+        "from textwrap import dedent\n",
+        "\n",
+        "def construct_confidence_prompt(prompt, response):\n",
+        "    template = f\"\"\"Question:\n",
+        "        {prompt}\n",
+        "        Answer: {response}\n",
+        "        {CONFIDENCE_PROMPT}\"\"\"\n",
+        "    return dedent(template)\n",
+        "\n",
+        "\n",
+        "def parse_self_eval_score(text: str):\n",
+        "    try:\n",
+        "        score = int(text.split(\"Score:\")[-1].strip())\n",
+        "        return (score - 1) / 4  # Normalize to 0-1 range\n",
+        "    except:\n",
+        "        return np.nan"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Creating a function to run all benchmarks on the dataset:\n",
+        "\n",
+        "Now that we've instantiated all of our hallucination detection models, we can create a function to run all of them on a given dataset subsection and extract the scores.\n",
+        "\n",
+        "By creating a function, we can run this on arbitrary slices of the dataset, which is helpful for debugging and testing."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 20,
+      "metadata": {
+        "id": "9ltqxAGG3J9V"
+      },
+      "outputs": [],
+      "source": [
+        "import pandas as pd\n",
+        "from deepeval import evaluate as deepeval_evaluate\n",
+        "from deepeval.test_case import LLMTestCase\n",
+        "from deepeval.dataset import EvaluationDataset\n",
+        "\n",
+        "from datetime import datetime\n",
+        "\n",
+        "\n",
+        "def score_dataset_with_metrics(in_data: pd.DataFrame):\n",
```

Comment:
[nit] imo it would be a lot more readable to separate each metric (or at least groups of them, eg. TLM/RAGAS/deepval) into their own helper functions then only call those helpers in this huge function. It'll make it nicer if users only want to run a subset of the evals

Reply:
Ok - done!